code/_core/obj/item/bullet/_bullet_cartridge.dm

```diff
@@ -209,8 +209,8 @@
 			var/obj/item/attachment/undermount/gun/AG = G.attachment_undermount
 			G = AG.stored_gun
 
-		if(G.bullet_time > 0)
-			PROGRESS_BAR(caller,G,G.bullet_time,/obj/item/weapon/ranged/bullet/proc/accept_bullet,caller,src)
+		if(G.bullet_time[src.type] > 0)
```

Comment:
You should add a list length check here. before the G.bullet_time[src.type] comparison.

Reply:
`if(G.bullet_time && G.bullet_time[src.type] > 0)`
Is this what you mean?